src/firefly/java/edu/caltech/ipac/firefly/server/util/DsvToDataGroup.java

```diff
@@ -33,14 +27,53 @@
  *
  * @author loi
  * @version $Id: DsvToDataGroup.java,v 1.2 2012/10/23 18:37:22 loi Exp $
+ *
+ * 09/28/17
+ * LZ added another method to sue InputStream to read data
+ *
  */
 public class DsvToDataGroup {
-    private static final Logger.LoggerImpl LOG = Logger.getLogger();
 
+    public static DataGroup parse(InputStream inf, CSVFormat format) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inf), IpacTableUtil.FILE_IO_BUFFER_SIZE);
+        return getData(reader, format);
+    }
     public static DataGroup parse(File inf, CSVFormat format) throws IOException {
 
         BufferedReader reader = new BufferedReader(new FileReader(inf), IpacTableUtil.FILE_IO_BUFFER_SIZE);
+        return getData(reader, format);
+
+       /* List<DataType> columns = new ArrayList<DataType>();
+        CSVParser parser = new CSVParser(reader, format);
```

Comment:
You should go ahead and remove this block of commented code.

Reply:
OK